File: HouseholdBudgetManager/src/HouseholdBudgetManager.WebApp/projects/household-budget-manager/src/app/models/expense-category.ts
export enum ExpenseCategory {
  Housing = 0,
  Transportation = 1,
  Food = 2,
  Healthcare = 3,
  Entertainment = 4,
  PersonalCare = 5,
  Education = 6,
  DebtPayment = 7,
  Savings = 8,
  Other = 9
}

export const EXPENSE_CATEGORY_LABELS: Record<ExpenseCategory, string> = {
  [ExpenseCategory.Housing]: 'Housing',
  [ExpenseCategory.Transportation]: 'Transportation',
  [ExpenseCategory.Food]: 'Food',
  [ExpenseCategory.Healthcare]: 'Healthcare',
  [ExpenseCategory.Entertainment]: 'Entertainment',
  [ExpenseCategory.PersonalCare]: 'Personal Care',
  [ExpenseCategory.Education]: 'Education',
  [ExpenseCategory.DebtPayment]: 'Debt Payment',
  [ExpenseCategory.Savings]: 'Savings',
  [ExpenseCategory.Other]: 'Other'
};

File: HouseholdBudgetManager/src/HouseholdBudgetManager.Core/Models/ExpenseAggregate/Enums/ExpenseCategory.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

namespace HouseholdBudgetManager.Core;

/// <summary>
/// Represents the category of an expense.
/// </summary>
public enum ExpenseCategory
{
    /// <summary>
    /// Housing expenses (rent, mortgage, utilities).
    /// </summary>
    Housing = 0,

    /// <summary>
    /// Transportation expenses.
    /// </summary>
    Transportation = 1,

    /// <summary>
    /// Food and groceries.
    /// </summary>
    Food = 2,

    /// <summary>
    /// Healthcare expenses.
    /// </summary>
    Healthcare = 3,

    /// <summary>
    /// Entertainment and recreation.
    /// </summary>
    Entertainment = 4,

    /// <summary>
    /// Personal care.
    /// </summary>
    PersonalCare = 5,

    /// <summary>
    /// Education expenses.
    /// </summary>
    Education = 6,

    /// <summary>
    /// Debt payments.
    /// </summary>
    DebtPayment = 7,

    /// <summary>
    /// Savings and investments.
    /// </summary>
    Savings = 8,

    /// <summary>
    /// Other expenses.
    /// </summary>
    Other = 9,
}
